### PR TITLE
fix: reserved function name should be private

### LINKED ISSUE #7253: Invalid variable assignment name change in ES5 with eval
### Describe the bug

The compiled code assigns a variable that doesn't exist `bmRnd1`. The variable declaration isn't renamed to match the assignment. I ran into this with `lottie-web@5.7.12` (build/player/lottie.js file). Upgrading that dependency fixed the problem (bmRnd is just a function definition in the updated code).

### Input code

```typescript
var bmRnd;
function a() {
  bmRnd = function () {};
}
eval('');
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "ecmascript",
      "jsx": false
    },
    "target": "es5",
  }
}
```

### Playground link

https://play.swc.rs/?version=1.3.49&code=H4sIAAAAAAAAAz2KTQqAIBBG93OKb6eeYegSLdpbagQ2gqmb8O4JQsv302zGfq%2FimChUOcqVBDlVcZuN1T86RHsavIS5YcG%2F6WbjTED2pWbBMDy4M3UiP0grZfgDK1lJ%2FmUAAAA%3D&config=H4sIAAAAAAAAA6vmUlBQyipOVrJSqAYygZyCxKLi1CI4HyhSXJlXklgBFFFKTc5NLE4uyiwoUdKByWYVg6TSEnOKU8FCtRAZpZLEovTUErCuYlOw8lquWgDsxhJabwAAAA%3D%3D

### Expected behavior

swc should rename (or not rename?) both the variable declaration and the variable assignment to `bmRnd`.

### Actual behavior

swc renames the variable assignment to a different name from the variable declaration. This results in a runtime error for the built code.

### Version

1.3.46

### Additional context

I'm not 100% sure if this is expected in ES5 with eval (maybe it is?).

### PR DESCRIPTION
**Pull request by hyf0**
**Description:**

Fixes #7253

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

**Comment by hyf0** [comment]
I wonder why anonymous class expression doesn't have this problem.🤔

https://play.swc.rs/?version=1.3.49&code=H4sIAAAAAAAAAytLLFJIyg3KS7Hm4korzUsuyczPUyjKL81LCUvMKU0t1kjLSUzXVKjmUoAoU7BVSM5JLC5WqK7lquXiSi1LzNFQV9e0BgA%2BVasUSAAAAA%3D%3D&config=H4sIAAAAAAAAA6vmUlBQyipOVrJSqAYygZyCxKLi1CI4HyhSXJlXklgBFFFKTc5NLE4uyiwoUdKByWYVg6TSEnOKU8FCtRAZpZLEovTUErCuYlOw8lquWgDsxhJabwAAAA%3D%3D

**Comment by hyf0** [comment]
sync up: I'm going to keep investigating this Sunday.

File: crates/swc_ecma_transforms/tests/es2015_function_name.rs [hunk 1]
```diff
@@ -57,7 +57,7 @@ test!(
     r#"number = function (x) {
   return x;
 };"#,
-    r#"number = function number(x) {
+    r#"number = function number1(x) {
```
Comment (hyf0) [inline crates/swc_ecma_transforms/tests/es2015_function_name.rs:60]: @kdy1 

After I used `private_ident!`, the problem of #2753 is solved. But the change in this test case is unexpected.

Hygiene Renamer would rename `number#private` to `number1` for these is an unresolved `number` in the global scope.

I wonder if Hygiene Renamer needs to apply special behaviors on top-level and non-top-level variables.

Comment (kdy1) [reply crates/swc_ecma_transforms/tests/es2015_function_name.rs:60]: I think the ideal behavior is to check if there's a reference for `number#global` in the scope of `number#private` and rename only if there are usages for both, but I'm not sure how we can implement it.

Comment (hyf0) [reply crates/swc_ecma_transforms/tests/es2015_function_name.rs:60]: Do you consider these test changes unacceptable? If not, we could merge this PR first after the CI is passed.

Comment (kdy1) [reply crates/swc_ecma_transforms/tests/es2015_function_name.rs:60]: I think it's fine

**Review by kdy1 — APPROVED** [review]
swc-bump:
 - swc_ecma_transforms_compat

**Pull request merged**